### DISCUSSION TITLE
ESP32-CAM and ADC

### DISCUSSION THREAD

**Discussion by davefes** [micropython/micropython#15559] [ESP32] [answered]
On the ESP32 you have to use ADC1 if you are using WiFi.  Which pins are available on an ESP32-CAM, in Micropython, to use as an ADC.
RandomNerdTutorials say:

> All these GPIOs are RTC and support ADC: GPIOs 2, 4, 12, 13, 14, and 15.

However, on the ESP32 all of these are ADC_2.

External ADC?

**Comment by scruss** [comment, marked as answer]
Looks like external ADC is the only way, if there are any pins left to hook something up

**Reply by davefes** [reply]
I will need to test ...  that if WiFi is not connected at the time of measurement then ADC_2 would be OK to use?  I don't need both running at the same time.

**Comment by bixb922** [comment]
Here is the complete list of pins https://docs.espressif.com/projects/esp-idf/en/stable/esp32/api-reference/peripherals/gpio.html

There should be ADC 1 pins available, but that depends on the board you are using.

**Reply by davefes** [reply]
I think for the ESP32_CAM (AiThinker clone) ADC_1 pins are all used for the camera or SD card.  Any how they are not brought-out to the headers..
